tensorflow/core/kernels/dml_conv_ops.cc

```diff
@@ -1347,4 +1348,598 @@ class DmlDepthwiseConv2DBackpropInputKernel : public DmlKernel {
+    OP_REQUIRES(context, dilations.size() == 5,
+                errors::InvalidArgument("Dilation rates field must "
+                                        "specify 5 dimensions"));
+
+    int32 dilation_n = GetTensorDim(dilations, data_format, 'N');
+    int32 dilation_c = GetTensorDim(dilations, data_format, 'C');
+    dilation_d = GetTensorDim(dilations, data_format, '0');
+    dilation_h = GetTensorDim(dilations, data_format, '1');
+    dilation_w = GetTensorDim(dilations, data_format, '2');
+
+    OP_REQUIRES(context, (dilation_n == 1 && dilation_c == 1),
+                errors::InvalidArgument(
+                    "Current implementation does not yet support "
+                    "dilation rates in the batch and depth dimensions."));
+    OP_REQUIRES(
+        context, (dilation_d > 0 && dilation_h > 0 && dilation_w > 0),
+        errors::InvalidArgument("Dilated rates should be larger than 0."));
+    OP_REQUIRES_OK(context, context->GetAttr("padding", &padding));
+  }
+
+  TensorFormat data_format;
+  Padding padding;
+  int32 stride_d;
+  int32 stride_h;
+  int32 stride_w;
+  int32 dilation_d;
+  int32 dilation_h;
+  int32 dilation_w;
+};
+
+class Conv3DInitHelper : public InitializationHelper {
+ public:
+  using Attributes = Conv3DAttributes<true>;
+
+  Conv3DInitHelper(OpKernelContext* context,
+                   std::shared_ptr<const Attributes> attr)
+      : attr_(attr) {
+    // Input tensor is of the following dimensions:
+    // [ batch, in_z, in_y, in_x, in_channels ]
+    const Tensor& input = context->input(0);
+
+    // Input filter is of the following dimensions:
+    // [ filter_z, filter_y, filter_x, in_channels, out_channels]
+    const Tensor& filter = context->input(1);
+
+    // NOTE: The ordering of the spatial dimensions is arbitrary, but has to be
+    // kept consistent between input/filter/output.
+    OP_REQUIRES(context, input.dims() == 5,
+                errors::InvalidArgument("input must be 5-dimensional"));
+    OP_REQUIRES(context, filter.dims() == 5,
+                errors::InvalidArgument("filter must be 5-dimensional"));
+
+    const int64 in_batch = GetTensorDim(input, attr->data_format, 'N');
+    const int64 in_channels = GetTensorDim(input, attr->data_format, 'C');
+    const int64 in_depth = GetTensorDim(input, attr->data_format, '0');
+    const int64 in_height = GetTensorDim(input, attr->data_format, '1');
+    const int64 in_width = GetTensorDim(input, attr->data_format, '2');
+
+    const int64 filter_depth = filter.dim_size(0);
+    const int64 filter_height = filter.dim_size(1);
+    const int64 filter_width = filter.dim_size(2);
+    const int64 filter_channels = filter.dim_size(3);
+    const int64 out_channels = filter.dim_size(4);
+
+    OP_REQUIRES(context, in_channels % filter_channels == 0,
+                errors::InvalidArgument(
+                    "Input depth must be evenly divisible by filter depth: ",
```

Comment:
Same typo here, channel count, not depth, as the filter `in_depth` can be less than or equal to the input depth.

Reply:
See other comment for same resolution.